resources/puppetlabs/lein-ezbake/template/global/controller.sh.erb

```diff
@@ -0,0 +1,135 @@
+################
+build_os=$1
+build_ver=$2
+
+BASEPATH='pkg_artifacts'
+mkdir $BASEPATH
+if [ ! -d "$BASEPATH/base" ]; then
+        DESTDIR="$BASEPATH/base" bash install.sh install_redhat
+fi
+if [ ! -d "$BASEPATH/systemd_el" ]; then
+        cp -r "$BASEPATH/base" "$BASEPATH/systemd_el"
+        DESTDIR="$BASEPATH/systemd_el" bash install.sh systemd_redhat
+fi
+if [ ! -d "$BASEPATH/old_el" ]; then
+        cp -r "$BASEPATH/base" "$BASEPATH/old_el"
+        DESTDIR="$BASEPATH/old_el" bash install.sh sysv_init_redhat
+fi
+if [ ! -d "$BASEPATH/old_sles" ]; then
+        cp -r "$BASEPATH/base" "$BASEPATH/old_sles"
+        DESTDIR="$BASEPATH/old_sles" bash install.sh sysv_init_suse
+fi
+
+# things are only different if we have docs, deb docs get
+# installed in an unversioned folder but rpm docs get installed
+# in a versioned folder.
+if [ -d ext/docs ]; then
+        if [ ! -d "$BASEPATH/base_deb" ]; then
+                DESTDIR="$BASEPATH/base_deb" bash install.sh install_deb
+                if [ ! -d "$BASEPATH/systemd_deb" ]; then
+                        cp -r "$BASEPATH/base_deb" "$BASEPATH/systemd_deb"
+                        DESTDIR="$BASEPATH/systemd_deb" bash install.sh systemd_deb
+                fi
+                if [ ! -d "$BASEPATH/sysvinit_deb" ]; then
+                        cp -r "$BASEPATH/base_deb" "$BASEPATH/sysvinit_deb"
+                        DESTDIR="$BASEPATH/sysvinit_deb" bash install.sh sysv_init_deb
+                fi
+        fi
+else
+        if [ ! -d "$BASEPATH/systemd_deb" ]; then
+                cp -r "$BASEPATH/base" "$BASEPATH/systemd_deb"
+                DESTDIR="$BASEPATH/systemd_deb" bash install.sh systemd_deb
+        fi
+        if [ ! -d "$BASEPATH/sysvinit_deb" ]; then
+                cp -r "$BASEPATH/base" "$BASEPATH/sysvinit_deb"
+                DESTDIR="$BASEPATH/sysvinit_deb" bash install sysv_init_deb
+        fi
+fi
+
+os=$build_os
+if [ "$os" = "debian" ];
+        os_dist=$build_ver
+else
+        os_version=$build_ver
+fi
+
+case $os in
+        # there's no differences in packaging for deb vs ubuntu
+        # if that changes we'll need to fix this
+        debian|ubuntu)
+                if [ "$os_dist" = 'trusty' ]; then
+                        dir="$BASEPATH/sysvinit_deb"
+                else
+                        dir="$BASEPATH/systemd_deb"
+                fi
+                ;;
+        el)
+                if [ "$os_version" -gt '6' ]; then
+                        dir="$BASEPATH/systemd_el"
+                else
+                        dir="$BASEPATH/old_el"
+                fi
+                ;;
+        sles)
+                if [ "$os_version" -gt '11' ]; then
+                        dir="$BASEPATH/systemd_el"
+                else
+                        dir="$BASEPATH/old_sles"
+                fi
+                ;;
+        *)
+                echo "I have no idea what I'm doing with $os, teach me?" >&2
+                exit 1
+                ;;
+esac
```

Comment:
@melissa is this case statement less awful? We can easily pass in the version and os from the rake file where we call this, but I sorta feel like since the directories are all created in this script, making something else know what they are feels messy.